src/openloco/things/CreateVehicle.cpp

```diff
@@ -436,9 +436,15 @@ namespace openloco::things::vehicle
         Vehicle train(head);
         // lastVeh will point to the vehicle component prior to the tail (head, unk_1, unk_2 *here*, tail) or (... bogie, bogie, body *here*, tail)
         openloco::vehicle* lastVeh = nullptr;
-        if (train.cars.size() > 0)
+        if (!train.cars.empty())
         {
-            lastVeh = reinterpret_cast<openloco::vehicle*>(train.cars.back().carComponents.back().body);
+            for (auto& car : train.cars)
```

Comment:
Unfortunately this change had to be made. It sort of reverts it back to how the original assembly worked. This function is not called very often so performance is not critical.